h2o-jetty-9/src/main/java/water/webserver/jetty9/Jetty9WebsocketServlet.java

```diff
@@ -0,0 +1,76 @@
+package water.webserver.jetty9;
+
+
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.WebSocketListener;
+import org.eclipse.jetty.websocket.servlet.WebSocketServlet;
+import org.eclipse.jetty.websocket.servlet.WebSocketServletFactory;
+import water.webserver.iface.WebsocketConnection;
+import water.webserver.iface.WebsocketHandler;
+import water.webserver.iface.WebsocketServlet;
+
+import java.io.IOException;
+
+public class Jetty9WebsocketServlet extends WebSocketServlet {
+
+    private final WebsocketServlet impl;
+
+    public Jetty9WebsocketServlet(WebsocketServlet impl) {
+        this.impl = impl;
+    }
+
+    static class Jetty9WebsocketConnection implements WebsocketConnection {
+
+        private final Session sess;
+
+        Jetty9WebsocketConnection(Session sess) {
+            this.sess = sess;
+        }
+
+        @Override
+        public void sendMessage(String message) throws IOException {
```

Comment:
Would it make sense to primarily use the asynchronous API that enables you to receive a response in a straightforward way?

https://www.eclipse.org/jetty/javadoc/9.4.30.v20200611/org/eclipse/jetty/websocket/api/RemoteEndpoint.html#sendStringByFuture(java.lang.String)

Reply:
Do we need the ACK method?

Reply:
ad async API: there is no request-response in websockets, but I agree we could make this API easier to work with in the case we require every message to be confirmed with a ACK response, which I am not sure I want to do

the ACK helper is steam specific, this part of the code is completely independent, for now the hello-ack messages are an easy way to test the comunication and I think something like this could be useful in the future, but I dont think I will want to keep ACK the way it is now

Reply:
I like how this is enforced with the interface introduced. And it's steam specific, it's part of the contract. This is completely fine to me. 

Req/resp has to be implemented on top of WS exactly like that. There are implementations and standarda for that, e.g. JAX-RS and RestEasy/Jersey. Those are however quite heavy.

Reply:
Not that much heavy, but I like the minimalist approach in H2O more.